### ISSUE TITLE
Traceback and crash (null pointer exception) when right-clicking to "Edit" a tag in the sidebar a second time

### ISSUE THREAD

**Issue by nekohayo** [getting-things-gnome/gtg#411]
In the flatpak version, but not when launched with `launch.sh` in the git version, when you right-click a tag in the sidebar and select "Edit", the app instantly crashes / shuts down like this:

```
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/GTG/gtk/browser/simple_color_selector.py", line 108, in on_expose
    self.__draw(cr)
  File "/app/lib/python3.7/site-packages/GTG/gtk/browser/simple_color_selector.py", line 71, in __draw
    Gdk.cairo_set_source_color(gdkcontext, my_color)
TypeError: Argument 1 does not allow None as a value

Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/GTG/gtk/browser/simple_color_selector.py", line 108, in on_expose
    self.__draw(cr)
  File "/app/lib/python3.7/site-packages/GTG/gtk/browser/simple_color_selector.py", line 71, in __draw
    Gdk.cairo_set_source_color(gdkcontext, my_color)
TypeError: Argument 1 does not allow None as a value

(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GTG+gtk+browser+simple_color_selector+SimpleColorSelectorPaletteItem': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GtkBox': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GtkBox': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GtkAlignment': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GTG+gtk+browser+simple_color_selector+SimpleColorSelector': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GtkAlignment': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GtkBox': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GtkBox': NULL pointer
(gtg:2): Gtk-WARNING **: 10:43:10.798: drawing failure for widget 'GTG+gtk+browser+tag_editor+TagEditor': NULL pointer
python3: ../../src/cairo.c:524: cairo_destroy: Assertion `CAIRO_REFERENCE_COUNT_HAS_REFERENCE (&cr->ref_count)' failed.
```

**Comment by diegogangl** [comment]
Can't confirm this one, did you do something else before? Is this Wayland? Which distro?

**Comment by nekohayo** [comment]
Didn't do something in particular before encountering the problem, happens on any tag, on Fedora 32, with both Wayland and Xorg. I think on Xorg it seemed to crash "instantly", whereas on my laptop running Wayland it shows the tag editor window briefly and _then_ crashes.

**Comment by diegogangl** [comment]
I'm on F32, Xorg as well. Can you try with an empty dataset?

**Comment by nekohayo** [comment]
I've encountered this again today, with the Git version started with launch.sh on my own data set, this time on Ubuntu 20.04 (Xorg GNOME session). It creates the exact same traceback and segfault as above. Since I ran launch.sh -d, I got some additional debug info just before the traceback:

```
2020-11-01 14:53:42,197 - DEBUG - main_window:on_tag_treeview_button_press_event:815 - Received button event #3 at 93, 419
2020-11-01 14:53:44,159 - ERROR - cell_renderer_tags:do_render:162 - Can't load icon 'emblem-generic': gtk-icon-theme-error-quark: Icon 'emblem-generic' not present in theme Adwaita (0)
2020-11-01 14:53:44,171 - ERROR - cell_renderer_tags:do_render:162 - Can't load icon 'emblem-urgent': gtk-icon-theme-error-quark: Icon 'emblem-urgent' not present in theme Adwaita (0)
2020-11-01 14:53:50,494 - DEBUG - main_window:on_tag_treeview_button_press_event:815 - Received button event #3 at 74, 629
2020-11-01 14:53:50,905 - ERROR - cell_renderer_tags:do_render:162 - Can't load icon 'emblem-favorite': gtk-icon-theme-error-quark: Icon 'emblem-favorite' not present in theme Adwaita (0)
```

...but I don't think this problem is actually related to the missing emblem icons in Ubuntu, because it also still occurs on my Fedora 32 GNOME Xorg session with the 0.4 flatpak package.

Testing further, I found out that what I thought was a race condition heisenbug is actually a solid "bohrbug": it does not happen with the "Bryce" dataset, but it happens with my personal dataset, and it happens only when you trigger the dialog for a second time, no matter which tag you're doing this with. So if I right-click and choose "Edit..." on tag foobar, the tag editor dialog successfully appears without crashing; if I close that dialog, then right-click and choose "Edit..." again, then I get the crash everytime.

**Comment by Neui** [comment]
I looked it quite some time ago. Here, `gdkcontext` is `None`, but it is passed as a parameter by an callback which from what I've looked (and remember) should always be non-`None`. The following messages makes me think that something has internally go very wrong for it to do that. I'd use something like [rr](https://rr-project.org/) to reproduce and record and check when that happens, but since I can't reproduce it and sharing that recording would also include the personal data, this is hard to debug.
